x/participation/keeper/msg_participate_test.go

```diff
@@ -34,23 +41,72 @@ func Test_msgServer_Participate(t *testing.T) {
 	require.NotNil(t, res.BaseAuction)
 	auctionID := res.BaseAuction.Id
 
+	addrWithDels := sample.Address()
+	tk.DelegateN(sdkCtx, addrWithDels, 100, 10)
+	availableAlloc, err := tk.ParticipationKeeper.GetAvailableAllocations(sdkCtx, addrWithDels)
+	require.NoError(t, err)
+
 	tests := []struct {
-		name string
-		msg  *types.MsgParticipate
-		err  error
+		name                  string
+		msg                   *types.MsgParticipate
+		desiredUsedAlloc      uint64
+		currentAvailableAlloc uint64
+		err                   error
+		untypedErr            bool
 	}{
 		{
-			name: "should allow to add the participant as allowed bidder in the auction",
+			name: "valid message",
+			msg: &types.MsgParticipate{
+				Participant: addrWithDels,
+				AuctionID:   auctionID,
+				TierID:      1,
+			},
+			desiredUsedAlloc:      1,
+			currentAvailableAlloc: availableAlloc,
+		},
+		{
+			name: "should prevent if double bid from same address",
+			msg: &types.MsgParticipate{
+				Participant: addrWithDels,
+				AuctionID:   auctionID,
+				TierID:      1,
+			},
+			err: types.ErrInvalidBidder,
+		},
+		{
+			name: "should prevent if address is invalid",
+			msg: &types.MsgParticipate{
+				Participant: "invalid",
+				AuctionID:   auctionID,
+				TierID:      1,
+			},
+			untypedErr: true,
+			err:        errors.New("decoding bech32 failed: invalid bech32 string length 7"),
```

Comment:
We don't need a testcase for invalid address, invalid address must always be handled in the message static `Validate` method

Also I would prefer not asserting exact error because it can be hard to maintain even more when the error is not returned from our codebase in this case